source/contributing/implementation/datamanipulator.rst

```diff
@@ -522,6 +524,75 @@ can be registered by subsequent calls of the ``registerValueProcessor()`` method
     dataRegistry.registerValueProcessor(Keys.MAX_HEALTH, new MaxHealthValueProcessor());
 
 
+Implementing Block Data
+=======================
+
+Block data is somewhat different from other types of data in that it is implemented by mixing in to the block itself.
+There are several methods in ``org.spongepowered.mixin.core.block.MixinBlock`` that must be overridden to implement
+data for blocks.
+
+.. code-block:: java
+    
+    @Mixin(BlockHorizontal.class)
+    public abstract class MixinBlockHorizontal extends MixinBlock {
+
+        [...]
+    }
+
+``supports()`` should return ``true`` if either the ``ImmutableDataManipulator`` interface is assignable from the
+``Class`` passed in as the argument, or the superclass supports it.
+
+.. code-block:: java
+
+    @Override
+    public boolean supports(Class<? extends ImmutableDataManipulator<?, ?>> immutable) {
+        return super.supports(immutable) || ImmutableDirectionalData.class.isAssignableFrom(immutable);
+    }
+
+``getStateWithData()`` should return a new ``BlockState`` with the data from the ``ImmutableDataManipulator`` applied
+to it. If the manipulator is not directly supported, the method should delegate to the superclass.
+
+.. code-block:: java
+
+    @Override
+    public Optional<BlockState> getStateWithData(IBlockState blockState, ImmutableDataManipulator<?, ?> manipulator) {
+        if (manipulator instanceof ImmutableDirectionalData) {
+            final Direction direction = ((ImmutableDirectionalData) manipulator).direction().get();
+            final EnumFacing facing = DirectionResolver.getFor(direction);
+            return Optional.of((BlockState) blockState.withProperty(BlockHorizontal.FACING, facing));
+        }
+        return super.getStateWithData(blockState, manipulator);
+    }
+
+``getStateWithValue()`` is the equivalent of ``getStateWithData()``, but works with single ``Key``s.
```

Comment:
You need to escape the backtick and add a space to pluralise, eg. ``DataManipulator``\ s
I think that's what breaks the formatting.